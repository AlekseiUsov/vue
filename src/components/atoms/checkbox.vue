<template>
    <label :class="$style.label">
        <input type="checkbox" :class="$style.inputCheckbox" />
        <span :class="$style.checkbox"></span>
        <span :class="$style.checkboxText"></span>
    </label>
</template>

<script>
export default {
    name: "checkbox"
};
</script>

<style lang="scss" module>
@import "@/assets/main.scss";

.label {
    display: flex;
    align-items: center;
}

.inputCheckbox {
    opacity: 0;
}

.checkboxText {
    color: $brown;
}

.checkbox {
    display: block;
    width: 1rem;
    height: 1rem;
    border: 1px solid $yellow;
    border-radius: 0.2em;
    transition: 0.2s;
    margin-right: 0.5rem;
}

.inputCheckbox:checked+.checkbox {
    background-image: url('../../assets/checked.svg');
    background-size: contain;
    background-color: $yellow;
}
</style>