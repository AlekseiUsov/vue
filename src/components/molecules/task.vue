<template>
    <li :class="$style.task">
        <checkbox />
        <close />
    </li>
</template>

<script>
import checkbox from '@/components/atoms/checkbox';
import close from '@/components/atoms/close';


export default {
    components: {
        checkbox,
        close,
    }
};
</script>

<style lang="scss" module>
@import "@/assets/main.scss";

.task {
    display: flex;
    justify-content: space-between;
    padding: 0.25rem;
    box-sizing: border-box;
    border-radius: 0.5rem;
    background-color: $light-beige;
    margin-top: 1rem;
}

.task:first-child {
    margin-top: 0;
}
</style>