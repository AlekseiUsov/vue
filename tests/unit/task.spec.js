import { mount } from "@vue/test-utils";
import checkbox from "../../src/components/atoms/ToDoCheckbox.vue";
import close from "../../src/components/atoms/ToDoCloseBth.vue";
import task from "../../src/components/molecules/ToDoTask.vue";

describe("ToDoTask.vue", () => {
  it("тестируем чекбокс", async () => {
    const wrapper = mount(task);
    expect(wrapper).toMatchSnapshot();
  });

  it("тестируем пропсы task", async () => {
    const wrapper = mount(task, {
      propsData: {
        title: "String",
        isChecked: true,
      },
    });
    expect(wrapper.find("li").text()).toBe("String");
    expect(wrapper.findComponent(checkbox).props().isChecked).toBe(true);
  });

  it("тестируем клик по чекбоксу в таске", async () => {
    const wrapper = mount(task);

    await wrapper.findComponent(checkbox).vm.$emit("checkCheckbox");
    expect(wrapper.emitted("changeStatus")).toBeTruthy();
  });

  it("тестируем клик по кнопке close в таске", async () => {
    const wrapper = mount(task);

    await wrapper.findComponent(close).vm.$emit("close");
    expect(wrapper.emitted("closeTask")).toBeTruthy();
  });
});
