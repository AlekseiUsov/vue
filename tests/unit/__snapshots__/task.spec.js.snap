// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoTask.vue тестируем чекбокс 1`] = `
<li><label><input type="checkbox"> <span></span></label>

  <div><span></span></div>
</li>
`;
